wpimath/src/main/native/include/frc/util/PIDConstants.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+namespace frc {
+
+/** A class to configure a PIDController with */
+struct PIDConstants {
+ public:
+  // Factor for "proportion" control
+  const double Kp = 0.0;
+  // Factor for "integral" control
+  const double Ki = 0.0;
+  // Factor for "derivative" control
+  const double Kd = 0.0;
```

Comment:
If these will be const as they are in the java implementation(final), then they cannot be initialized to zero.

Reply:
Should I add a default constructor value instead?

Reply:
A default constructor seems like a strange thing to have, I would just not have any kind of default value at all.

Reply:
If this type were an aggregate of non-const fields instead, no constructors would be needed, and designated initializers would let users, say, skip initializing the rarely used Ki. https://en.cppreference.com/w/cpp/language/aggregate_initialization

The amount of code this saves seems worth deferring contract checks until it's given to a PIDController.

Reply:
Yeah that's probably the best way around this. 

I cannot remember off the top of my head, but would a constexpr constructor still be possible if the variables are not const? 

I think it would be useful to have so any pid constants instances can be constexpr such as in a Constants.h file.

Reply:
Yes, members are implicitly const in a constexpr object.